types/index.d.ts

```diff
@@ -543,6 +553,31 @@ interface ConnectOptions {
   models: string | Bone[];
 }
 
+interface InitOptions {
+  underscored?: boolean;
+  tableName?: string;
+  hooks?: {
+    [key in 'beforeCreate' | 'beforeBulkCreate' | 'beforeUpdate' | 'beforeSave' |  'beforeUpsert' | 'beforeRemove' ]: (options: QueryOptions) => Promise<void>
+  } | {
+    [key in 'afterCreate' | 'afterBulkCreate' | 'afterUpdate' | 'afterSave' | 'afterUpsert' | 'afterRemove' ]: (instance: Bone, result: Object) => Promise<void>
+  };
```

Comment:
added before/after names

It seems typescript removed `AsyncFunction`, added specific function type definitions. Static hooks not added yet.